Hotelreservation

subpal opened the pull request:
Added Hotel Reservation Project

subpal committed:
Added hotelreservation

subpal committed:
Added Hotel Reservation

subpal committed:
Update README.md

Himanshu-it commented on HotelReservation/src/main/java/Day.java:
```diff
@@ -0,0 +1,11 @@
+    WEEKEND(1), WEEKDAY(2);
+    private int id;
+
+    Day(int value){
```
Why have you added id to Day? Enums are just  constants. Further there does not seem any requirement of adding a private variable id.

subpal replied:
We need a integer value for each enum, it is used for overriding hashcode function in Request class.

Himanshu-it replied:
Okay. But in that case it is better to write a wrapper class which contains an id and a type as Day
and make that class as the key of hashmap.

Himanshu-it commented on HotelReservation/src/test/java/HotelTests.java:
```diff
@@ -0,0 +1,119 @@
+import java.util.HashMap;
+
+public class HotelTests {
+    private static HotelList myHotels = new HotelList();
```
make these variables private. Even make setDefaultsHotel() public without static

Himanshu-it reviewed (commented):
Try to group java files in different packages. for example Customer, Days etc can go to Models, and HotelFinder in a service package.
You can think of some diff packages also

Commit:
Removed static from variables of HotelTests.java

shgupta22 commented on HotelReservation/src/main/java/TypeOfCustomer.java:
```diff
@@ -0,0 +1,12 @@
+enum Type{
```
Rename it to CustomerType. its misleading as Type could be anything

shgupta22 commented on HotelReservation/src/main/java/Request.java:
```diff
@@ -0,0 +1,31 @@
+import com.sun.corba.se.impl.encoding.TypeCodeOutputStream;
+
+public class Request {
```
You have added lambok dependency in pom but you are not using that. By using that you don't need to write getter setter equal and hashcode function by your own

shgupta22 commented on HotelReservation/src/main/java/HotelList.java:
```diff
@@ -0,0 +1,43 @@
+        listOfHotels.remove(h);
+    }
+
+    public int size(){
```
change this api to get the count of hotels, size is not clarifying what size it is

shgupta22 commented on HotelReservation/src/main/java/HotelList.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    public Boolean contains(String h){
+        for(int i = 0; i < listOfHotels.size(); ++i){
```
you don't need to explicitly iterate over the list. use api provided by java https://www.tutorialspoint.com/java/util/arraylist_contains.htm

subpal replied:
This function just takes a string `h` as hotelname but, the arraylist contains hotel objects which are not comparable with string `h`.

shgupta22 commented on HotelReservation/src/main/java/HotelList.java:
```diff
@@ -0,0 +1,43 @@
+
+    public Hotel getHotel(String name){
+        for(int i = 0; i < listOfHotels.size(); ++i){
+            if(listOfHotels.get(i).getHotelName().equals(name)){
```
same as above https://www.tutorialspoint.com/java/util/arraylist_contains.htm

Commit:
Added model and service package

shgupta22 commented on HotelReservation/src/main/java/HotelFinder.java:
```diff
@@ -0,0 +1,47 @@
+import java.util.*;
+
+class SortArgument {
```
Don't write multiple classes in a single file. Create different files for each class

shgupta22 commented on HotelReservation/src/main/java/Hotel.java:
```diff
@@ -0,0 +1,26 @@
+import java.util.HashMap;
+
+public class Hotel {
```
use lambok for boilerplate code

shgupta22 reviewed (changes requested):
There are lots of changes required, we will discuss them later. Apart from that, you need to change a lot of things. Please work on those.

Commit:
1. Renamed Customer class to Request.
2. Renamed Request class to Category.

Commit:
1. Changed the name of function size to count in HotelList class.
2. Created separate files for class Comp and SortArgument.

Commit:
1. Renamed SortArgment class to HotelSortArguement.
2. Created a new package utils for HotelSortArgument.
1. Renamed SortArgment class to HotelSortArguement.
2. Created a new package utils for HotelSortArgument.
1. Renamed SortArgment class to HotelSortArguement.
2. Created a new package utils for HotelSortArgument.
1. Renamed SortArgment class to HotelSortArguement.
2. Created a new package utils for HotelSortArgument.
1. Renamed SortArgment class to HotelSortArguement.
2. Created a new package utils for HotelSortArgument.
1. Renamed SortArgment class to HotelSortArguement.
2. Created a new package utils for HotelSortArgument.
1. Renamed SortArgment class to HotelSortArguement.
2. Created a new package utils for HotelSortArgument.
1. Renamed SortArgment class to HotelSortArguement.
2. Created a new package utils for HotelSortArgument.
1. Renamed SortArgment class to HotelSortArguement.
2. Created a new package utils for HotelSortArgument.
1. Renamed SortArgment class to HotelSortArguement.
2. Created a new package utils for HotelSortArgument.
1. Renamed SortArgment class to HotelSortArguement.
2. Created a new package utils for HotelSortArgument.

Commit:
Renamed variables to proper names and created utils package for comparator

Commit:
Moved HotelSortArgument from Request class to its own file.

Commit:
Added Maths Operations Project

Commit:
Used lombok annotations

Commit:
Removed IDs from enums
